packages/material-ui/src/CircularProgress/CircularProgress.js

```diff
@@ -1,92 +1,151 @@
-  colorPrimary: {
-    color: theme.palette.primary.main,
-  },
-  /* Styles applied to the root element if `color="secondary"`. */
-  colorSecondary: {
-    color: theme.palette.secondary.main,
-  },
-  /* Styles applied to the svg element. */
-  svg: {
-    display: 'block', // Keeps the progress centered
-  },
-  /* Styles applied to the `circle` svg path. */
-  circle: {
-    stroke: 'currentColor',
-    // Use butt to follow the specification, by chance, it's already the default CSS value.
-    // strokeLinecap: 'butt',
-  },
-  /* Styles applied to the `circle` svg path if `variant="determinate"`. */
-  circleDeterminate: {
-    transition: theme.transitions.create('stroke-dashoffset'),
-  },
-  /* Styles applied to the `circle` svg path if `variant="indeterminate"`. */
-  circleIndeterminate: {
-    animation: '$circular-dash 1.4s ease-in-out infinite',
-    // Some default value that looks fine waiting for the animation to kicks in.
-    strokeDasharray: '80px, 200px',
-    strokeDashoffset: '0px', // Add the unit to fix a Edge 16 and below bug.
-  },
-  '@keyframes circular-rotate': {
-    '0%': {
-      transform: 'rotate(0deg)',
-    },
-    '100%': {
-      transform: 'rotate(360deg)',
-    },
-  },
-  '@keyframes circular-dash': {
-    '0%': {
-      strokeDasharray: '1px, 200px',
-      strokeDashoffset: '0px',
-    },
-    '50%': {
-      strokeDasharray: '100px, 200px',
-      strokeDashoffset: '-15px',
-    },
-    '100%': {
-      strokeDasharray: '100px, 200px',
-      strokeDashoffset: '-125px',
+const circularRotateKeyframe = keyframes`
+0% {
+  transform: rotate(0deg);
+}
+100% {
+  transform: rotate(360deg);
+}
+`;
+
+const circularDashKeyframe = keyframes`
+0% {
+  strokeDasharray: 1px, 200px;
+  strokeDashoffset: 0px;
+}
+50% {
+  strokeDasharray: 100px, 200px;
+  strokeDashoffset: -15px;
+}
+100% {
+  strokeDasharray: 100px, 200px;
+  strokeDashoffset: -125px;
+}
+`;
+
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...styles[styleProps.variant],
+    ...styles[`color${capitalize(styleProps.color)}`],
+    [`& .${circularProgressClasses.svg}`]: styles.svg,
+    [`& .${circularProgressClasses.circle}`]: {
+      ...styles.circle,
+      ...styles[`circle${capitalize(styleProps.variant)}`],
+      ...(styleProps.disableShrink && styles.circleDisableShrink),
     },
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, variant, color, disableShrink } = styleProps;
+
+  const slots = {
+    root: ['root', variant, `color${capitalize(color)}`],
+    svg: ['svg'],
+    circle: ['circle', `circle${capitalize(variant)}`, disableShrink && 'circleDisableShrink'],
+  };
+
+  return composeClasses(slots, getCircularProgressUtilityClass, classes);
+};
+
+const CircularProgressRoot = experimentalStyled(
+  'span',
+  { shouldForwardProp: (prop) => shouldForwardProp(prop) || prop === 'classes' },
```

Comment:
```suggestion
  {},
```

Is there a valid use case for the `classes` prop?